content/blogposts/2022/code-intelligence-platform.md

```diff
@@ -0,0 +1,33 @@
+---
+title: "From code search toward a code intelligence platform"
+publishDate: 
+description: TODO
+tags: [blog]
+slug: "code-search-to-code-intelligence"
+published: false
+heroImage: 
+socialImage: 
+changelogItems:
+---
+
+> NOTE: This post is very WIP. To be edited and tightened up a lot.
+
+No matter how good your other dev tools, coding is still complex. Even if you've configured your editor just right, sped up your test suite, set up continuous deployment, added observability, etc., coding is still complex. Higher-level languages, libraries, frameworks, cloud providers, etc., that encapsulate complexity just let you spend your complexity budget elsewhere. The overall complexity of code has increased, not declined, over time, and there's no sign of that trend changing.
+
+As consumers of software, we (mostly) benefit from this phenomenon in the form of better technology in our everyday lives.
+
+But as developers of software, we all feel the pain of code complexity: the struggle to understand big codebases, not being able to get in flow, encountering tech debt that turns a tiny fix into a multi-day ordeal, all the wasted time, etc.
+
+Code search is helpful in tackling complexity, but it's not enough. Code search is for finding things *in* code (matching lines in files). What helps you find things *about* code? Let's call it a code intelligence platform. Compared to code search, it's more helpful, more times per day. This is like how Google went from helping you find things *on* the web (matching web pages), to helping you find things *from* the web (information and answers). Here are some examples `TODO: make it clear which these are available and which are future directions`:
```

Comment:
Comment from @felixfbecker (in the [previous draft gdoc](https://docs.google.com/document/d/1EAXfMBidVX1WJERdtnU8x-urrjpnc-80-PoOzeQPNpw/edit#)):

> This alone doesn't really justify the name "platform" to me (making it appear to me as a dev and unfamiliar reader as a marketing buzzword without substance). A platform is something that you can build other things on top of. I think if we want to show and explain that we're becoming a code intelligence platform, we need to talk about why we're sunsetting our extension API (they didn't deliver on the platform promise, because they were UI only) and what our future vision is that is soon to come over the 4.x releases (metadata, ownership, ...).

> Maybe we can even talk about some things that are shipped in 4.0 that show that direction/vision. E.g. is the repo metadata by the search team going to be ready to be talked about? That's simple, but it clearly shows the difference between UI-only extensions and persisted data that becomes available to search queries, to the GraphQL API, code monitors, to search aggregations (also shipping in 4.0), ...

> That paints the picture of an actual "platform" to me (you can imagine how you could eventually build entire tools like coverage tracking on top of Sourcegraph by uploading the data into Sourcegraph, using it in queries, dashboards, code monitors, ...)

Reply:
I think there's an opportunity here to articulate that this is a platform because we are building new products and use cases on top of the core search technology.

There are different ideas around of what defines a "platform", but for us, all of our product suite is built with a hard dependency on Search technology - which is a big part of it being a platform for us.